tap_snapchat_ads/streams.py

```diff
@@ -384,9 +430,12 @@ def sync_endpoint(
                     # API request data
                     data = {}
                     try:
-                        data = client.get(
-                            url=next_url,
-                            endpoint=stream_name)
+                        # checks if profiles are selected, if not extracts data for all orgs and ad_accounts
+                        if config.get('org_account_ids', []) and stream_name in ['organizations', 'ad_accounts']:
+                            data = self.extract_selected_profile_data(config, client, data_key_array, parent_id)
+                            data['request_status'] = 'SUCCESS'
+                        else:
+                            data = client.get(url=next_url, endpoint=stream_name)
```

Comment:
Earlier API calls are made with `next_url` and it has some querysting and many query parameters stuff but now only a simple URL is used for API calls, so does pagination will still get used or those parts of code will never be utilized in API calls now?
Next URL Reference: https://github.com/singer-io/tap-snapchat-ads/blob/16fa29dd2e4c32555840fd14a5b0afab90c555cc/tap_snapchat_ads/streams.py#L412-L425

Reply:
thanks @savan-chovatiya for reviewing..
when the stream name is either organisation or ad_account then we call extract_selected_profile_data(being called inside if condition) method which extracts data for all the IDs that are mentioned in config.json.
So pagination is not required for organisation and ad_account streams but it is still being used for other streams(in else condition)

Reply:
@kethan1122 Okay got it. I thought the `else` part was not usable now.